releasing/run-goreleaser.sh

```diff
@@ -100,9 +100,6 @@ builds:
 
   ldflags: >
     -s
-    -X sigs.k8s.io/kustomize/api/provenance.version={{.Version}}
-    -X sigs.k8s.io/kustomize/api/provenance.gitCommit={{.Commit}}
-    -X sigs.k8s.io/kustomize/api/provenance.buildDate={{.Date}}
```

Comment:
We at least do need to keep the date flag. Based on the notes I left on this PR, it seems possible the others always work from the BuildInfo source, in which case we can keep them deleted (but we need to confirm). Perhaps we can add any that we're keeping to the Makefile so that local builds are more sane anyway.